dvc/repo/data.py

```diff
@@ -234,10 +251,21 @@ def status(repo: "Repo", untracked_files: str = "no", **kwargs: Any) -> Status:
     git_info = _git_info(repo.scm, untracked_files=untracked_files)
     untracked = git_info.get("untracked", [])
     untracked = _transform_git_paths_to_dvc(repo, untracked)
+
+    entries_not_in_remote = (
+        _get_entries_not_in_remote(
+            repo,
+            granular=granular,
+            remote_refresh=remote_refresh,
+        )
+        if not_in_remote
+        else []
+    )
```

Comment:
Can you please move this after `_diff_index_to_wtree` operation? I missed this.

When dvc switches to Git `HEAD` (or, switches back) as it does here in `_diff_head_to_index`, cached `repo.index` gets reset, which means we are re-reading `dvc.yaml`/`.dvc`/`dvc.lock` files again.